server/tests/e2e/federated.spec.ts

```diff
@@ -136,17 +131,15 @@ describe("end-to-end federated", function () {
 
     const aggregator = aggregators.getAggregator(lusCovidTask, trainingScheme)
     const client = new clients.federated.FederatedClient(url, lusCovidTask, aggregator)
-    const disco = new Disco(lusCovidTask, { scheme: trainingScheme, client })
+    const disco = await Disco.fromTask(lusCovidTask, client, { scheme: trainingScheme })
 
     const logs = List(await arrayFromAsync(disco.trainByRound(data)));
     await disco.close()
 
-    if (aggregator.model === undefined) {
-      throw new Error('model was not set')
-    }
     const validationLogs = logs.last()?.epochs.last()?.validation
     expect(validationLogs?.accuracy).to.be.greaterThan(0.6)
-    return aggregator.model.weights
+
+    return disco.trainer.model.weights
```

Comment:
what do you think of adding a getter to disco to simplify this expression? Like `disco.weights`

Reply:
weights are not really supposed to be exposed to the outside, only the model is relevant to users, so it should be a bit hard to get. at first, I didn't even wanted the trainer to be available (but some tests needed it).
I can meet you halfway by exposing `disco.model`, WDYT?
note: in my mind, `Disco` is an easy access object, for newcomers and not-so-ML people. if people want more control, they can use the trainer directly.

Reply:
Oh I see no need to change it then. I was thinking this would be the API to run inference with models but it's through the validator actually.